go/cagra/search_params.go

```diff
@@ -0,0 +1,160 @@
+package cagra
```

Comment:
For consistency, I think I'd prefer to mirror the top-level namspaces also (e.g. `cuvs/neighbors/cagra`. Can we do nested packages like this?